Update devices/tuya.ts

Giexperience opened the pull request:
Add a new device GX04

Koenkk commented on src/devices/tuya.ts:
```diff
@@ -13124,6 +13124,25 @@ const definitions: DefinitionWithExtend[] = [
         },
     },
+    {
+        fingerprint: tuya.fingerprint('TS0601', ['_TZE284_nhgdf6qr']),
```
Since the definition is the same, could you add this to the `TS0601_soil` definition and use `tuya.whitelabel` for correct detections (search in this file for some examples)

Giexperience commented:
Hello. I noticed that they indeed have the same definition. However, the brand and model are different, and the product appearance also varies. I was wondering if it could be treated as a new product? Or is there another way to map TZE284_nhgdf6qr to the correct image, brand, and model?


| |
GIEX-CINLIN-RAUL
|
|
***@***.***
|
---- Replied Message ----
| From | Koen ***@***.***> |
| Date | 11/27/2024 15:14 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [Koenkk/zigbee-herdsman-converters] Update devices/tuya.ts (PR #8383) |

@Koenkk commented on this pull request.

In src/devices/tuya.ts:

> @@ -13124,6 +13124,25 @@ const definitions: DefinitionWithExtend[] = [
             ],
         },
     },
+    {
+        fingerprint: tuya.fingerprint('TS0601', ['_TZE284_nhgdf6qr']),


Since the definition is the same, could you add this to the TS0601_soil definition and use tuya.whitelabel for correct detections (search in this file for some examples)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

Giexperience commented:
Okay, I understand. I will modify it.

Giexperience commented:
Hello, I have added the code for the new device GX04 at line 2158 according to your guidance, and I have also added the new device GX03 at the end. Please review it.

Koenkk commented on src/devices/tuya.ts:
```diff
@@ -2155,6 +2155,7 @@ const definitions: DefinitionWithExtend[] = [
                 [15, 'battery', tuya.valueConverter.raw],
             ],
         },
+        whiteLabel: [tuya.whitelabel('GIEX', 'GX04', 'Soil Moisture Sensor', ['_TZE284_nhgdf6qr'])],
```
Also update `fingerprint: tuya.fingerprint('TS0601', ['_TZE284_aao3yzhs']),`

Koenkk commented on src/devices/tuya.ts:
```diff
@@ -13148,6 +13149,46 @@ const definitions: DefinitionWithExtend[] = [
+        ],
+        meta: {
+            tuyaDatapoints: [
+                [1, 'valve1', tuya.valueConverter.onOff],
```
```suggestion
                [1, 'valve_1', tuya.valueConverter.onOff],
```

Koenkk commented on src/devices/tuya.ts:
```diff
@@ -13148,6 +13149,46 @@ const definitions: DefinitionWithExtend[] = [
+                [1, 'valve1', tuya.valueConverter.onOff],
+                [59, 'battery', tuya.valueConverter.raw],
+                [104, 'valve1', tuya.valueConverterBasic.lookup({OFF: tuya.enum(2), ON: tuya.enum(0)})],
+                [2, 'valve2', tuya.valueConverter.onOff],
```
```suggestion
                [2, 'valve_2', tuya.valueConverter.onOff],
```

Koenkk commented on src/devices/tuya.ts:
```diff
@@ -13148,6 +13149,46 @@ const definitions: DefinitionWithExtend[] = [
+                [104, 'valve1', tuya.valueConverterBasic.lookup({OFF: tuya.enum(2), ON: tuya.enum(0)})],
+                [2, 'valve2', tuya.valueConverter.onOff],
+                [105, 'valve1', tuya.valueConverterBasic.lookup({OFF: tuya.enum(2), ON: tuya.enum(0)})],
+                [13, 'countdown_V1', tuya.valueConverter.raw],
```
```suggestion
                [13, 'countdown_1', tuya.valueConverter.raw],
```

Giexperience commented:
Thanks

Koenkk commented:
Thanks!
